### PR TITLE
test: Add unit and integration tests

### PR DESCRIPTION
**Pull request by bvestli**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,3 +30,45 @@ See the `README.md` under each module's subfolder for a list of supported inputs
...
+### Prerequisites
+
+* You need to have Terraform installed (version: see [versions.tf](modules/bucket/versions.tf))
+* You need to have Go installed (version >=1.19 because of the `go:build` directive format used in the go files)
+* You need to have gcloud installed and be authenticated (`gcloud auth application-default login`)
+* Your user must have the necessary permissions on the GCP project used to run the tests
+
+### Versioning
+
+This repository is versioned. Please see GitHub´s [documentation](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository#creating-a-release) describing how to create a new release.
```
Comment (sindrel) [inline README.md:55]: Versioning and release workflows are automated using Release Please, perhaps we should point to their docs?
https://github.com/googleapis/release-please#release-please

File: tests/cloud_memorystore_integration_test.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	// Create client
+	ctx := context.Background()
+	client, err := redis.NewCloudRedisClient(ctx)
+	if err != nil {
+		t.Fail()
+		fmt.Println("ERROR:: ", err.Error())
+	}
+	defer client.Close()
+
+	reqInstance := "projects/ent-tfmodules-dev/locations/europe-west1/instances/" + expectedName
```
Comment (sindrel) [inline tests/cloud_memorystore_integration_test.go:65]: Perhaps this string could be parameterized and/or set from environment/project variables?

File: tests/fixtures/redis/variables.tf [hunk 3]
```diff
@@ -0,0 +1,27 @@
+variable "init" {
```
Comment (sindrel) [inline tests/fixtures/redis/variables.tf:1]: Should we omit this variable, as it's not used? And just rely on testing against a live init module? Cc @Spikra

**Review by sindrel — CHANGES_REQUESTED** [review]
I have a few comments, but overall this looks great. I like that you've included more details on how to develop and run tests etc.

**Comment by bvestli** [comment]
Thank you for good review, @sindrel. The suggested improvements is now implemented.

**Review by sindrel — APPROVED** [review]

**Pull request merged**